src/libbson/src/bson/bson-decimal128.c

```diff
@@ -634,7 +634,7 @@ bson_decimal128_from_string_w_len (const char *string,     /* IN */
       /* Shift exponent to significand and decrease */
       last_digit++;
 
-      if (last_digit - first_digit > BSON_DECIMAL128_MAX_DIGITS) {
+      if (last_digit - first_digit >= BSON_DECIMAL128_MAX_DIGITS) {
```

Comment:
If `last_digit - first_digit` is exactly `34` (`BSON_DECIMAL128_MAX_DIGITS`), we want to treat that as out-of-bounds as well.